legend-engine-authentication-pure/src/main/resources/core_authentication/authentication/metamodel.pure

```diff
@@ -0,0 +1,87 @@
+// Copyright 2022 Goldman Sachs
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//      http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+import meta::pure::authentication::*;
+import meta::pure::credential::*;
+
+Class <<typemodifiers.abstract>> meta::pure::credential::Credential
+{
+  provider: CredentialProvider[1];
+}
+
+Class meta::pure::credential::PlainTextCredential extends Credential
+{
+}
+
+Class meta::pure::credential::PrivateKeyCredential extends Credential
+{
+}
+
+Class meta::pure::credential::JsonWebTokenCredential extends Credential
+{
+}
+
+Class <<typemodifiers.abstract>> meta::pure::credential::CredentialProvider
+{
+}
+
+Class meta::pure::credential::VaultCredentialProvider extends CredentialProvider
+{
+  vaultReference: String[1];
+}
+
+Class meta::pure::credential::OidcCredentialProvider extends CredentialProvider
+{
+  discoveryUrl: String[1];
+}
+
+Class meta::pure::credential::OAuthTokenProvider extends CredentialProvider
+{
+  url: String[1];
+}
+
+Class <<typemodifiers.abstract>> meta::pure::authentication::Authentication
```

Comment:
```Authentication``` as a generic concept makes sense.  It's how a platform component (stores, persistence, mastery etc) can describe authentication behavior.

In some cases authentication is disabled. For e.g a database used for testing. We can either support this by making ```Authentication``` an optional attribute of ```Connection```